File: src/main/java/io/github/alexiscomete/lapinousecond/commands/CommandInServer.java
package io.github.alexiscomete.lapinousecond.commands;

import io.github.alexiscomete.lapinousecond.Main;
import io.github.alexiscomete.lapinousecond.Player;
import io.github.alexiscomete.lapinousecond.save.SaveManager;
import org.javacord.api.entity.server.Server;
import org.javacord.api.event.message.MessageCreateEvent;

import java.util.Optional;

public abstract class CommandInServer extends CommandWithAccount {

    public CommandInServer(String description, String name, String totalDescription, String... perms) {
        super(description, name, totalDescription, perms);
    }

    @Override
    public void execute(MessageCreateEvent messageCreateEvent, String content, String[] args, Player p) {

        Optional<Server> servOp = messageCreateEvent.getServer();
        if (servOp.isPresent()) {
            Server serv = servOp.get();
            if (serv.getId() == p.getServer()) {
                executeC(messageCreateEvent, content, args, p);
            } else {
                messageCreateEvent.getMessage().reply("Utilisez cette commande dans un salon du serveur actuel : " + p.getServer());
            }
        } else {
            messageCreateEvent.getMessage().reply("Utilisez cette commande dans un salon du serveur actuel : " + p.getServer());
        }

    }

    public abstract void executeC(MessageCreateEvent messageCreateEvent, String content, String[] args, Player p);
}

File: src/main/java/io/github/alexiscomete/lapinousecond/commands/CommandInServer.kt
package io.github.alexiscomete.lapinousecond.commands

import io.github.alexiscomete.lapinousecond.entity.Player
import org.javacord.api.event.message.MessageCreateEvent

abstract class CommandInServer(description: String, name: String, totalDescription: String, vararg perms: String) :
    CommandWithAccount(description, name, totalDescription, *perms) {
    override fun execute(messageCreateEvent: MessageCreateEvent, content: String, args: Array<String>, p: Player) {
        val servOp = messageCreateEvent.server
        if (servOp.isPresent) {
            val serv = servOp.get()
            if (serv.id == p["serv"].toLong()) {
                executeC(messageCreateEvent, content, args, p)
            } else {
                messageCreateEvent.message.reply("Utilisez cette commande dans un salon du serveur actuel : " + p["serv"].toLong())
            }
        } else {
            messageCreateEvent.message.reply("Utilisez cette commande dans un salon du serveur actuel : " + p["serv"].toLong())
        }
    }

    abstract fun executeC(messageCreateEvent: MessageCreateEvent, content: String, args: Array<String>, p: Player)
}